docs/contributing/create-a-starter.md

```diff
@@ -1,9 +1,74 @@
 ---
 title: Create a Starter
-issue: https://github.com/gatsbyjs/gatsby/issues/8103
 ---
 
-This is a stub. Help our community expand it.
+[Starters](/docs/starters/) are boilerplate projects that Gatsby developers can use to set up a new site quickly. Before creating a starter, it may be helpful to peruse the [Gatsby Starter Library](/starters/) to see what already exists and determine how your starter will provide value.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Basic requirements
+
+For a starter to work properly, it needs to include some files (see the [Hello World starter](https://github.com/gatsbyjs/gatsby-starter-hello-world/) for a barebones example):
+
+- `README.md`: instructions for how to install and configure your starter, a list of its features or structure, and any helpful tips.
+- `package.json`: the "command center" for Gatsby dependencies and scripts. Find an example in the [Hello World Starter](https://github.com/gatsbyjs/gatsby-starter-hello-world/blob/master/package.json).
+- `gatsby-config.js`: a space to add configurable data and plugins. See [Gatsby Config](/docs/gatsby-config/) for more information.
+- `src/pages`: a directory for page components to live, with at least [one example](https://github.com/gatsbyjs/gatsby-starter-hello-world/blob/master/src/pages/index.js).
+- `static`: a directory for static assets, such as a `favicon.ico` file.
+- `.gitignore` _(optional)_: a file telling Git which resources to leave out of source control, such as the `node_modules` directory, log files, Gatsby `.cache` and `public` directories.
```

Comment:
I wouldn't make this optional as it's really bad practice to check-in the node_modules and/or public folder.